### PR TITLE
HW9

### PR DESCRIPTION
**Pull request by iarkov**
Homework #9: Strings and Arrays

### CODE REVIEW

File: functionsHW9.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <iostream>
+#include <cctype>
```
Comment (bohdanpc) [inline functionsHW9.h]: no includes in headers unless you really need them here for you functions prototypes

File: functionsHW9.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+void translateArray(int numbers[], int size) {
+    for (int i = 0; i < size; i++) {
+        if (numbers[i] > 0) {
+            std::cout << numbers[i] * 2 << " ";
+        }
+        else {
+            std::cout << 0 << " ";
+        }
+    }
```
Comment (bohdanpc) [inline functionsHW9.cpp]: not output but translation of original array

File: functionsHW9.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        if (islower(str[i])) {
+            std::cout << static_cast<char>(std::toupper(str[i]));
+        }
+        else {
+            std::cout << static_cast<char>(str[i]);
+        }
+    }
+}
+
+bool isPalindrom(const char str[]) {
```
Comment (bohdanpc) [inline functionsHW9.cpp:26]: `Паліндром -  слово, число, набір символів, що однаково читається в обох напрямках: зліва направо і справа наліво`
So, ' ' symbol is valid

File: functionsHW9.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    while (str[i] != ' ' && str[i] != '\0') {
+        bool isAlpha = std::isalpha(str[i]);
+        bool isVowel = std::tolower(str[i]) == 'a' ||
+            std::tolower(str[i]) == 'e' ||
+            std::tolower(str[i]) == 'o' ||
+            std::tolower(str[i]) == 'u' ||
+            std::tolower(str[i]) == 'i';
+
+        if (isAlpha && isVowel) {
```
Comment (bohdanpc) [inline functionsHW9.cpp]: I wonder what symbols are counted as Vowels but not Alpha's

File: functionsHW9.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        else if (isAlpha && !isVowel) {
+            consonantsCount++;
+        }
+
+        i++;
+    }
+}
+
+bool isEqual(const char str1[], const char str2[]) {
+    if (strlen(str1) == strlen(str2)) {
```
Comment (bohdanpc) [inline functionsHW9.cpp]: if (strlen(str1) != strlen(str2)) 
    return false;

int i = 0;
while (str1[i] != '\0') {
        if (str1[i] != str2[i]) {
            return false;
        }
}
    return false;

Looks much more neat, doesn't it?

Besides, this function doesn't work after all.

File: functionsHW9.cpp [hunk 6]
```diff
@@ -23,7 +25,7 @@ void toUpperCase(char str[]) {
 }
 
 bool isPalindrom(const char str[]) {
-    int wordLength = -1;
+    int wordLength = 0;
     for (int j = 0; j <= strlen(str); j++) {
         if (str[j] == ' ' || str[j] == '\0') {
             break;
```
Comment (helen-black) [inline functionsHW9.cpp]: int wordLength = strlen(str);

File: functionsHW9.cpp [hunk 7]
```diff
@@ -33,8 +35,12 @@ bool isPalindrom(const char str[]) {
...
-    for (int i = 0; i <= wordLength - i; i++) {
-        if (str[i] != str[wordLength - i]) {
+    if (!wordLength) {
+        return false;
+    }
+
+    for (int i = 0; i <= wordLength - 1 - i; i++) {
+        if (str[i] != str[wordLength - 1 - i]) {
             return false;
             break;
```
Comment (helen-black) [inline functionsHW9.cpp:44]: you don't need break after return

File: functionsHW9.cpp [hunk 8]
```diff
@@ -53,7 +59,7 @@ void parseStringLetters(const char str[], int& vowelsCount, int& consonantsCount
             std::tolower(str[i]) == 'u' ||
             std::tolower(str[i]) == 'i';
 
-        if (isAlpha && isVowel) {
+        if (isVowel) {
             vowelsCount++;
         }
         else if (isAlpha && !isVowel) {
```
Comment (helen-black) [inline functionsHW9.cpp:64]: you already know that isVowel == false

File: functionsHW9.cpp [hunk 9]
```diff
@@ -65,16 +71,17 @@ void parseStringLetters(const char str[], int& vowelsCount, int& consonantsCount
 }
 
 bool isEqual(const char str1[], const char str2[]) {
-    if (strlen(str1) == strlen(str2)) {
+    if (strlen(str1) != strlen(str2)) {
+        return false;
+    }
+    else {
```
Comment (helen-black) [inline functionsHW9.cpp:76]: you don't need else here

**Review by helen-black — CHANGES_REQUESTED** [review]

**Pull request merged**